### ISSUE TITLE
Potential overflow vulnerability detected

### ISSUE THREAD

**Issue by iamrohandhiman** [iamrohandhiman/Neural-Networks-from-scratch#7]
def train(network, loss, loss_prime, x_train, y_train, count=100, learning_rate=0.1):

    for e in range(count):
        error = 0
        for x, y in zip(x_train, y_train):
            output = predict(network, x)
            error += loss(y, output)
            grad = loss_prime(y, output)

            for layer in reversed(network):
                grad = layer.backward_propagation(grad, learning_rate)
        print(f"{e + 1}/{count}, error={error}")
